app-modules/knowledge-base/resources/views/filament/pages/badges.blade.php

```diff
@@ -33,7 +33,36 @@
 --}}
 
 <div class="flex gap-2">
-    <x-filament::badge>
-        {{ $getRecord()?->status?->name ?? '' }}
-    </x-filament::badge>
+    @if ($getRecord()->status)
+        <x-filament::badge>
+            {{ $getRecord()?->status?->name ?? '' }}
+        </x-filament::badge>
+    @endif
+    @if (!empty($getRecord()->public) && !empty($getRecord()->category_id))
+        <x-filament::badge
+            icon="heroicon-m-clipboard"
+            icon-position="after"
+        >
+            <button
+                type="button"
+                x-data
+                x-on:click="
+        window.navigator.clipboard.writeText(@js(route('portal.knowledge-management.show') . '/categories/' . $getRecord()->category_id . '/articles/' . $getRecord()->id)).then(() => {
+            $tooltip('Copied!', {
+                theme: $store.theme,
+                timeout: 2000,
+            })
+        }).catch(err => {
+            console.error('Failed to copy text: ', err);
+        })
+    "
+            >
+                {{ !empty($getRecord()->public) ? 'Public' : 'Internal' }}
```

Comment:
You have already confirmed that this is public above, so this can be simplified:

```suggestion
                Public
```